src/boards/include/boards/pimoroni_motor2040.h

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+// -----------------------------------------------------
+// NOTE: THIS HEADER IS ALSO INCLUDED BY ASSEMBLER SO
+//       SHOULD ONLY CONSIST OF PREPROCESSOR DIRECTIVES
+// -----------------------------------------------------
+
+#ifndef _BOARDS_PIMORONI_MOTOR2040_H
+#define _BOARDS_PIMORONI_MOTOR2040_H
+
+// For board detection
+#define PIMORONI_MOTOR2040
+
+// --- BOARD SPECIFIC ---
+#ifndef MOTOR2040_MOTOR_A_P_PIN
+#define MOTOR2040_MOTOR_A_P_PIN 4
+#endif
+
+#ifndef MOTOR2040_MOTOR_A_N_PIN
+#define MOTOR2040_MOTOR_A_N_PIN 5
+#endif
+
+#ifndef MOTOR2040_MOTOR_B_P_PIN
+#define MOTOR2040_MOTOR_B_P_PIN 6
+#endif
+
+#ifndef MOTOR2040_MOTOR_B_N_PIN
+#define MOTOR2040_MOTOR_B_N_PIN 7
+#endif
+
+#ifndef MOTOR2040_MOTOR_C_P_PIN
+#define MOTOR2040_MOTOR_C_P_PIN 8
+#endif
+
+#ifndef MOTOR2040_MOTOR_C_N_PIN
+#define MOTOR2040_MOTOR_C_N_PIN 9
+#endif
+
+#ifndef MOTOR2040_MOTOR_D_P_PIN
+#define MOTOR2040_MOTOR_D_P_PIN 10
+#endif
+
+#ifndef MOTOR2040_MOTOR_D_N_PIN
+#define MOTOR2040_MOTOR_D_N_PIN 11
+#endif
+
+#ifndef MOTOR2040_NUM_MOTORS
+#define MOTOR2040_NUM_MOTORS 4
+#endif
+
+#ifndef MOTOR2040_ENCODER_A_A_PIN
+#define MOTOR2040_ENCODER_A_A_PIN 0
+#endif
+
+#ifndef MOTOR2040_ENCODER_A_B_PIN
+#define MOTOR2040_ENCODER_A_B_PIN 1
+#endif
+
+#ifndef MOTOR2040_ENCODER_B_A_PIN
+#define MOTOR2040_ENCODER_B_A_PIN 2
+#endif
+
+#ifndef MOTOR2040_ENCODER_B_B_PIN
+#define MOTOR2040_ENCODER_B_B_PIN 3
+#endif
+
+#ifndef MOTOR2040_ENCODER_C_A_PIN
+#define MOTOR2040_ENCODER_C_A_PIN 12
+#endif
+
+#ifndef MOTOR2040_ENCODER_C_B_PIN
+#define MOTOR2040_ENCODER_C_B_PIN 13
+#endif
+
+#ifndef MOTOR2040_ENCODER_D_A_PIN
+#define MOTOR2040_ENCODER_D_A_PIN 14
+#endif
+
+#ifndef MOTOR2040_ENCODER_D_B_PIN
+#define MOTOR2040_ENCODER_D_B_PIN 15
+#endif
+
+#ifndef MOTOR2040_NUM_ENCODERS
+#define MOTOR2040_NUM_ENCODERS 4
+#endif
```

Comment:
Ah thanks. TBH I didn't fully understand the intention of some of these ifdefs when I first started adding boards, so have just been following the same template ever since 😅. Probably best for me to raise a separate PR to update all of them.

Reply:
Right. I believe the original intention of all these `#ifndef`s, was so that somebody could e.g. put a Raspberry Pi Pico in a breadboard, and if they happened to have a USB-UART adaptor connected to GPIO4 and GPIO5 they could do
```
#define PICO_DEFAULT_UART 1
#define PICO_DEFAULT_UART_RX_PIN 5
#define PICO_DEFAULT_UART_TX_PIN 4
```
and they'd still be able to use the rest of the code in e.g. https://github.com/raspberrypi/pico-examples/tree/master/hello_world/serial as-is, with no further modifications needed.
See also Section 2.6.2. and Appendix B in https://datasheets.raspberrypi.com/pico/raspberry-pi-pico-c-sdk.pdf

Reply:
Based on that description then, it sounds like many of the `#ifdefs` in our product board files should be removed

Reply:
yeah; the above comments are correct. if they are physical things then you don't need to #ifdef them, but really no harm

Reply:
Okay, I've removed all the unnecessary `#ifdefs` from the board file.